mmv1/third_party/terraform/tests/resource_google_service_account_iam_test.go.erb

```diff
@@ -216,6 +223,44 @@ func TestAccServiceAccountIamPolicy_withCondition(t *testing.T) {
 	})
 }
 
+func TestAccServiceAccountIamMember_federatedIdentity(t *testing.T) {
+	t.Parallel()
+
+	account := fmt.Sprintf("tf-test-%d", randInt(t))
+	pool := fmt.Sprintf("tf-test-%d", randInt(t))
+	poolProvider := fmt.Sprintf("tf-test-%d", randInt(t))
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccServiceAccountIamMember_federatedIdentity(account, pool, poolProvider),
+			},
+			{
+				ResourceName:      "google_service_account_iam_member.impersonate",
+				ImportStateIdFunc: testAccServiceAccountIamMember_generateFedereatedIdentityStateId,
```

Comment:
the member value is created with in import statement of 3 values.

serviceAccountId, role, and member. Member is composite import statement that is crafted with the `google_iam_workload_identity_pool` name. Since its generated at runtime is easiest just to extract this value directly from the existing state. Manually crafting it would require quite an exhaustive sprintf which would detract from readability imo

Reply:
If we don't specify `ImportStateIdFunc` the value used for the import test will be the implicit `id` field of the `ResourceName` resource. I'm not following why that doesn't work here?